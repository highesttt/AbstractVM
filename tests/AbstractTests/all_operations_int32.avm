; simple math
push int32(1)
push int32(2)
add
assert int32(3)
clear

push int32(1)
push int32(2)
sub
assert int32(-1)

push int32(-2)
mul
assert int32(2)

push int32(2)
div
assert int32(1)

push int32(2)
mod
dump

; math with numbers of a different type
clear
push float(1)
push int32(2)
add
assert float(3)

clear
push float(2)
push int32(1)
sub
assert float(1)

clear
push float(2)
push int32(1)
mul
assert float(2)

clear
push float(2)
push int32(1)
div
assert float(2)

clear
push float(2)
push int32(1)
mod
assert float(0)

dump

exit