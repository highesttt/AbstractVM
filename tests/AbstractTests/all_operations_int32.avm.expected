Return: 0
1
0
